### PR TITLE
Add Zcash to the onboarding supported network list

### PR DESCRIPTION
**Pull request by cypt4**
Add Zcash to the onboarding supported network list
 Add Zcash networks to the settings page
 Add wallet tests for Zcash
    
Resolves https://github.com/brave/brave-browser/issues/35732

Test plan:
With enabled Zcash feature:
1) Check Zcash is shown on the onboarding screen when wallet is created
image

2) Check that network list contains Zcash
image

3) Check onboarding checkbox is working correctly for Zcash
 
4) Check visibility settings are working correctly

With disabled Zcash feature:
1) Check there is no Zcash on the onboarding screen.
2) No Zcash in the networks list

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by yrliou — APPROVED** [review]

File: components/brave_wallet/common/common_utils_unittest.cc [hunk 1]
```diff
@@ -200,15 +230,30 @@ TEST(CommonUtils, MakeAccountId) {
...
+    if (keyring == mojom::KeyringId::kDefault) {
+      continue;
+    } else if (IsBitcoinKeyring(keyring) || IsZCashKeyring(keyring)) {
+      EXPECT_DCHECK_DEATH(MakeAccountId(id.coin, keyring, id.kind, id.address));
+    } else {
+      EXPECT_NE(
+          id.unique_key,
+          MakeAccountId(id.coin, keyring, id.kind, id.address)->unique_key);
+    }
+  }
```
Comment (supermassive) [inline components/brave_wallet/common/common_utils_unittest.cc:257]: Please return comments

File: components/brave_wallet/common/test_utils.cc [hunk 2]
```diff
@@ -14,7 +14,7 @@ namespace brave_wallet {
 bool AllCoinsTested() {
   // Change hardcoded value here only when all failed callers have adequate
   // testing for newly added coin.
-  return 4 == std::size(kAllCoins);
+  return 5 == std::size(kAllCoins) && 8 == std::size(kAllKeyrings);
```
Comment (supermassive) [inline components/brave_wallet/common/test_utils.cc:17]: I don't see changes in `value_conversion_utils_unittest.cc` and `account_resolver_delegate_impl_unittest.cc` but they touch AllCoinsTested

Comment (supermassive) [reply components/brave_wallet/common/test_utils.cc:17]: Why there is now also keyrings check? The idea was that number of coins is enough

File: components/brave_wallet/browser/brave_wallet_utils_unittest.cc [hunk 3]
```diff
@@ -735,6 +749,22 @@ TEST(BraveWalletUtilsUnitTest, GetAllChainsTest) {
...
+  ASSERT_EQ(zec_chains.size(), 2u);
+  EXPECT_EQ(zec_chains[0]->chain_id, mojom::kZCashMainnet);
+  EXPECT_EQ(zec_chains[0]->decimals, 123);
+  EXPECT_EQ(zec_chains[1]->chain_id, mojom::kZCashTestnet);
+  EXPECT_THAT(zec_chains[0]->supported_keyrings,
+              ElementsAreArray({mojom::KeyringId::kZCashMainnet}));
+  EXPECT_THAT(zec_chains[1]->supported_keyrings,
+              ElementsAreArray({mojom::KeyringId::kZCashTestnet}));
+
   EXPECT_TRUE(AllCoinsTested());
```
Comment (supermassive) [inline components/brave_wallet/browser/brave_wallet_utils_unittest.cc:768]: Why no change for the next test?

Comment (cypt4) [reply components/brave_wallet/browser/brave_wallet_utils_unittest.cc:770]: Added

File: components/brave_wallet/common/test_utils.cc (hunk 2 above)
Comment (cypt4) [reply components/brave_wallet/common/test_utils.cc:17]: Added

File: components/brave_wallet/common/common_utils_unittest.cc (hunk 1 above)
Comment (cypt4) [reply components/brave_wallet/common/common_utils_unittest.cc]: Done

File: components/brave_wallet/common/test_utils.cc (hunk 2 above)
Comment (cypt4) [reply components/brave_wallet/common/test_utils.cc:17]: > Why there is now also keyrings check? The idea was that number of coins is enough

This is an additional check to ensure keyrings list is updated along with coins list.

**Review by supermassive — COMMENTED** [review]
wallet core lgtm

File: components/brave_wallet/browser/brave_wallet_utils_unittest.cc [hunk 4]
```diff
@@ -804,6 +834,17 @@ TEST(BraveWalletUtilsUnitTest, GetNetworkURLTest) {
       GURL("https://test-btc.com"),
       GetNetworkURL(&prefs, mojom::kBitcoinMainnet, mojom::CoinType::BTC));
 
+  EXPECT_EQ(GURL("https://mainnet.lightwalletd.com:9067/"),
```
Comment (kdenhartog) [inline components/brave_wallet/browser/brave_wallet_utils_unittest.cc:837]: We should be proxying mainnet by default. Testnet we have a precedent for not requiring it on so no concerns with that.

**Review by kdenhartog — APPROVED** [review]
Approving since the concern I have should be addressed in a follow up since it's somewhat unrelated to this PR.

**Pull request merged**